### PR TITLE
nixos/gitea: add settings option

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/57671

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

ping @kolaente is this what you'd like to see?

### CODE REVIEW

**Review by kolaente — APPROVED** [review]
Looks great!

**Review by Ma27 — APPROVED** [review]

**Review by infinisil — APPROVED** [review]
Awesome, looking good!

**Comment by aanderse** [comment]
ping @pvgoran

File: nixos/modules/services/misc/gitea.nix [hunk 1]
```diff
@@ -435,9 +474,12 @@ in
...
     users.groups.gitea = {};
 
-    warnings = optional (cfg.database.password != "")
-      ''config.services.gitea.database.password will be stored as plaintext
-        in the Nix store. Use database.passwordFile instead.'';
+    warnings =
+      optional (cfg.database.password != "") ''
+        config.services.gitea.database.password will be stored as plaintext in the Nix store. Use database.passwordFile instead.'' ++
+      optional (cfg.extraConfig != null) ''
+        services.gitea.`extraConfig` is deprecated, please use services.gitea.`settings`.
```
Comment (Infinisil) [inline nixos/modules/services/misc/gitea.nix:481]: One tiny thing here is that I'd change `config.services.gitea.database.password` to `services.gitea.database.password` for consistency

Comment (aanderse) [reply nixos/modules/services/misc/gitea.nix:481]: I was planning to remove the `password` option as part of https://github.com/NixOS/nixpkgs/issues/24288 shortly after this PR.

Comment (Infinisil) [reply nixos/modules/services/misc/gitea.nix:481]: +1 to that!

**Comment by aanderse** [comment]
@GrahamcOfBorg test gitea.mysql gitea.postgres gitea.sqlite3

**Comment by pvgoran** [comment]
This is a straightforward improvement, I like it. I didn't review the code (only looked at it briefly) and didn't test it, though.

It lacks documentation, though. :) I mean, the description is just a link to another (draft) PR, and that PR is about a specific setting.

**Review by srhb — APPROVED** [review]

**Comment by aanderse** [comment]
I wasn't able to reproduce this error locally. Does anyone have any ideas on that?

**Comment by aanderse** [comment]
@GrahamcOfBorg test gitea.mysql
@GrahamcOfBorg test gitea.postgres
@GrahamcOfBorg test gitea.sqlite3

**Comment by aanderse** [comment]
I guess the failure was a glitch in the matrix :man_shrugging:

**Pull request merged**